temba/contacts/tasks.py

```diff
@@ -117,3 +123,97 @@ def check_elasticsearch_lag():
                 return False
 
     return True
+
+
+@nonoverlapping_task(track_started=True, name="block_deactivated_contacts_task")
+def block_deactivated_contacts_task():
+    email_subject = f"{timezone.now().strftime('%B %d, %Y')} - list of deactivated phone numbers."
+    email_text = """
+    Hi There!
+    We have prepared a list of contacts that were blocked because of deactivated phone number.
+    You can download it in the attached file.
+
+    Thanks,
+    The CommunityConnectLabs Team.
+    """
+
+    all_blocked_contacts = {}
+    for org in Org.objects.filter(is_active=True):
+        client = org.get_twilio_client()
+        if not client:
+            continue
+
+        try:
+            # get the link from twilio to download deactivated phone numbers
+            response = client.request("GET", "https://messaging.twilio.com/v1/Deactivations", {"Date": "2020-09-05"})
+            response_json = json.loads(response.text)
+            if not response.ok or not response_json.get("redirect_to"):
+                continue
+
+            # download and parse deactivated phone numbers
+            redirect_to = response_json.get("redirect_to")
+            response = requests.get(redirect_to)
+            if not response.ok:
+                continue
+            numbers = response.text.split("\n")
+
+            # get contacts from the db that have phone number deactivated and block them
+            formatted_numbers = ", ".join(map(lambda x: f"('{x}')", numbers[:3])) or "('')"
+            contacts_to_block = list(
+                Contact.objects.raw(
```

Comment:
Do you have an estimate about how long this query takes to run? I mean, `formatted_numbers` could be so many, right?

Reply:
The average amount of deactivated numbers during the last week is 230K, the max up to 450K.